src/devtools/client/webconsole/components/Output/Message.js

```diff
@@ -17,17 +17,6 @@ const CollapseButton = require("devtools/client/webconsole/components/Output/Col
 const MessageRepeat = require("devtools/client/webconsole/components/Output/MessageRepeat");
 const PropTypes = require("prop-types");
 const SmartTrace = require("devtools/client/shared/components/SmartTrace");
-const { getLocationKey } = require("devtools/client/debugger/src/utils/breakpoint");
-
-function getIsSecondaryHighlighted(hoveredPoint, message) {
-  if (!message?.frame || !hoveredPoint?.location) {
-    return false;
-  }
-
-  const keyOne = getLocationKey(hoveredPoint.location);
-  const keyTwo = getLocationKey(message.frame);
-  return keyOne == keyTwo;
-}
 
 class Message extends Component {
```

Comment:
I wonder if we can promote Message to a `PureComponent` and drop `shouldComponentUpdate`

https://reactjs.org/docs/react-api.html#reactpurecomponent

Reply:
I was wondering that too, but I decided to try with minimal changes for now. We're passing a lot of props to the message components that I think may trigger unnecessary renders. But we should revisit this when we refactor the code.